gui/src/components/LogsViewer.svelte

```diff
@@ -81,26 +88,36 @@
       logViewport.scrollTop = logViewport.scrollHeight;
     }
   });
+
+  const setFilterStrDebounced = debounce((filterStr: string) => {
+    setFilterStr(workspaceId, workspace, filterStr);
+  }, 250);
 </script>
 
 <section>
   <h1>Logs</h1>
   {#if hasData(state.events)}
-    <div class="log-table-overflow-wrapper">
-      <div class="log-table-container" bind:this={logViewport}>
-        <table>
-          {#each state.events.data as event (event.id)}
-            <tr class="log-entry" style={logStyleFromHash(event.log)}>
-              <td>{shortDate(event.timestamp)}</td>
-              <td>{friendlyName(event.log)}</td>
-              <td>
-                <FormattedLogMessage message={event.message} />
-              </td>
-            </tr>
-          {/each}
-        </table>
+    <div class="log-overflow-wrapper">
+      <div class="log-container" bind:this={logViewport}>
+        {#each state.events.data as event (event.id)}
```

Comment:
Why did this change from `<table>` to styled flexbox `<div>` structure?

The reason this was originally changed to a table is because it automatically aligns all the timestamps and colored process titles vertically, which I figure is important, and as far as I can tell this new structure would make those titles misaligned if they have different lengths.

In theory, we *can* get the same effect without `<table>`, by doing the following:

- Using the list of all processes which have `logs=visible`, pad the right hand side of their titles with enough spaces to make all of their title strings the same number of characters long (i.e. the length of the longest one of them all) in the `LogsViewer`. Since logs are monospaced this should work.
- In a similar process, pad and clamp the timestamp strings up to a certain length, so that they are all the same length (I believe they already are, but I'm not sure if the current formatting is guaranteed to do that)

Reply:
Yeah - you are right - they do end up misaligned. I ended up changing this to try to get the text wrapping working with the process label only displaying on the first line, but I changed that design decision and never backed out the flexbox stuff. I think that we may eventually want to go more fixed-width with the log name since the filtering means that as you filter, the column width can change based on the longest log name of the log events that match that filter. I'll back this change out and just do the line wrapping within the table (which I'm assuming is simple as removing the `whitespace: pre`).